src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -201,7 +201,7 @@ class primitive_inst {
     primitive_inst(network& network, program_node const& node, bool allocate_memory);
 
     network& _network;
-    program_node const& _node;
+    program_node const* _node;
```

Comment:
I think we'd better to have this as unique_ptr

Reply:
I tried unique_ptr and shared_ptr, but both raised a `double free or corruption (out)` exception at runtime.

Reply:
Maybe, it is passed through "not" shared_ptr argument in some point? 
Then the reference count might not be tracked properly.

Reply:
As you can see, the `node` is passed by reference in L201
`primitive_inst(network& network, program_node const& node, bool allocate_memory);`

Reply:
If arg passing type is changed from reference to pointer, null instance checking feature will be lost. For this reason, I restored arg passing type from pointer to reference. I'm not sure if there any specific reason to use shared_ptr instead of plain pointer.

Reply:
Based on your review, I changed the plain pointers to smart pointers.

Reply:
@e-ddykim @yeonbok I think there's nothing wrong with raw pointers if we use them correctly across the library. smart pointers always carry some info about ownership over the object and its lifetime, i.e. when you pass object as shared_ptr, it implies that method/class shares ownership over the object and may extend lifetime. In our case there is no reason for primitive_inst to have a shared ownership over program_node as program itself is responsible for lifetime of program_nodes. And similar thing for kernel_impl_params - it shouldn't control lifetime of program object, so shared_ptr is kind of misleading here. So my proposal is to stay with raw pointers and keep in mind that raw pointer means no ownership over object lifetime.

Reply:
@vladimir-paramuzov  In my thought, still we need to track the lifetime of the object as long as the primitive_inst refers to it through pointer though we have the ownership onto the program. For example the program_node object might be deleted by some reason by the program, while the primitive_inst still refers to it. Or the primitive_inst deletes the pointer by accident. How do you think?

Reply:
To my mind, if `program_node` can be deleted by accident, then we don't need ref/ptr at all. So probably the ideal solution is to remove `network <- program` and `primitive_inst <- program_node` dependencies at all and use program + program_nodes just to construct executable network with copy of all necessary things.

If we need an intermediate solution, then I'd prefer raw pointers as it has much smaller impact and doesn't change the behavior in terms of lifetime.

Reply:
I reverted smart pointers to raw pointers as @vladimir-paramuzov guided. Thank you.